### PR TITLE
[OMPIRBuilder] Fix collapse test case

### PR DESCRIPTION
**Pull request by DominikAdamski**

### CODE REVIEW

File: llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp [hunk 1]
```diff
@@ -4045,9 +4045,10 @@ OpenMPIRBuilder::applyWorkshareLoopTarget(DebugLoc DL, CanonicalLoopInfo *CLI,
   // We need to model loop body region as the function f(cnt, loop_arg).
   // That's why we replace loop induction variable by the new counter
   // which will be one of loop body function argument
-  for (auto Use = CLI->getIndVar()->user_begin();
-       Use != CLI->getIndVar()->user_end(); ++Use) {
-    if (Instruction *Inst = dyn_cast<Instruction>(*Use)) {
+  std::vector<User *> Users(CLI->getIndVar()->user_begin(),
```
Comment (skatrak) [inline llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp:4048]: Nit: Maybe it's better to use `llvm::SmallVector`?

**Review by skatrak — APPROVED** [review]
LGTM

File: llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp (hunk 1 above)
Comment (DominikAdamski) [reply llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp]: Done

**Review by dpalermo — APPROVED** [review]
LGTM

**Pull request merged**